testing/ProbeTest.md

```diff
@@ -0,0 +1,81 @@
+## How to write an Unit Test for Probe
+
+### Step I Test preparation: Initialize the mocked Rainbow, the mocked PortFactory and the GaugeDescription clase
+
+Here is an example for setting a probe test. This works for **all probes**. For different probe, you need to specify their different configuration yaml file.  Your should put the input file of ***probe*** under the folder <path-to-your-testing>/resource/ 
+
+```java
+private File tempInput = extractResource("/blackhole/probe-input.txt");
+public BlackholeProbeTest() throws IOException {
+}
+@Before
+public void setUp() {
+    RainbowMocker.injectRainbow();
+    IRainbowConnectionPortFactory mockedPortFactory = mockConnectionPortFactory();
```

Comment:
I think we should avoid static import here. Otherwise, the user may get confused at first glance. They may ask what mockConnectionPortFactory() is.